src/util/saveArticles.ts

```diff
@@ -49,6 +54,9 @@ export async function downloadFiles(fileStore: FileStore, zimCreator: ZimCreator
             let isFailed = false;
             try {
                 if (resp.result && resp.result.content) {
+                    if (isImageUrl(resp.path) && downloader.webp) {
+                        resp.path += '.webp'
+                    }
```

Comment:
I don't think this is the right approach because:
* It duplicates a test which is already done in `getCompressedBody()`
* It is based on the filename (which is less trustable than the `mime-type`)

I belive we should avoid that here totally by:
* allowing `getCompressedBody` to update the filename in the `resp`
* secure this new resp is returned in `getContentCb()`

Than way we test only one time, optimize content and change filename together. Less code, quicker and less error prone.

Reply:
@kelson42 the response being retruned here by getContentCb is not having the property resp.path, resp.path is set in DownloadBulk function. should I then also try to implement these changes?

Reply:
another point is `isMimeTypeimage()` passes true for svg as well and we are not converting svg to webp so we shouldn't concatenate anything here I can apply it in `downloadBulk()` instead of here that would be better